client/sftp_client.py

```diff
@@ -895,13 +892,13 @@ def refresh(self, node='/'):
 			exp_lines.insert(0, Gtk.TreeRowReference.new(model, path))
 			counter = 0
 		for path in exp_lines:
-			path = path.get_path()
+			path = path.get_path() if path.get_path() is not None else 0 # prevents NoneType error
```

Comment:
This should not be calling `get_path()` twice, that's terribly inefficient. This should be `path = path.get_path() or 0`, if `path.get_path()` returns `None` then the evaluate will lead to path being set to 0. This also seems to me that if `get_path()` returns None there might be another issue here and it may not make sense to process the node because setting it to 0 will effectively ignore the node argument.